src/accounts/get/get-accounts-command.js

```diff
@@ -0,0 +1,62 @@
+const Command = require('../../core/commands/command');
+const LoggerFactory = require('../../core/logging/logger-factory');
+const logger = LoggerFactory.getLogger(__filename);
+class GetAccountsCommand extends Command {
+  constructor(teamConfigService) {
+    super();
+    this.teamConfigService = teamConfigService;
+  }
+
+  configure(controller) {
+    controller.hears(
+      /accounts/,
+      'direct_message,direct_mention',
+      (bot, message) => {
+        return this.teamConfigService
+          .getAllAccountsSafeView(message.team, bot)
+          .then(allAccountsSafeView => {
+            let accountsString =
+              'These are the accounts in this workspace:\n' +
+              allAccountsSafeView
+                .map(item => {
+                  let channelNames =
+                    item.channels.length > 0
+                      ? ` This is the channel account for ` +
+                        item.channels
+                          .map(
+                            channel =>
+                              `<#${channel.channelId}|${channel.channelName}>`
+                          )
+                          .join(', ') +
+                        '.'
+                      : '';
+                  let isDefaultAccount = item.isDefault
+                    ? ' *This is the default workspace account.*'
+                    : '';
+                  return `• \`${item.accountAlias}\`: Slack alias for ${
+                    item.accountName
+                  }.${isDefaultAccount}${channelNames}\n`;
+                })
+                .join('');
+            bot.reply(
+              message,
+              allAccountsSafeView.length === 0
+                ? "You haven't added any accounts yet. To add one, type @Alice add account"
+                : accountsString
+            );
+          })
```

Comment:
it took me a few reads to understand what going on here:
maybe try to separate functionality from param extraction
maybe something similar to this (didn't run it so probably need some work):

  configure(controller) {
    controller.hears(
      /accounts/,
      'direct_message,direct_mention',
      this.ask,
    );
  }

  ask(bot, message) {

    const getAllAccountSafeView = (allAccountsSafeView) => {
      let response;
      if (!allAccountsSafeView.length) {
        response = "You haven't added any accounts yet. To add one, type @Alice add account";
      } else {
        const accounts = this.fetchAccountsFromSafeView(allAccountsSafeView);
        response = `These are the accounts in this workspace:
          ${accounts.join('')}`;
      }
      return response;
    }

    return this.teamConfigService
      .getAllAccountsSafeView(message.team, bot)
      .then(getAllAccountSafeView)
      .then(response => bot.reply(message, response))
      .catch(err => logger.error(err));
  }

  fetchAccountsFromSafeView(allAccountsSafeView) {

   function mapFetchAccountsFromSafeView({ channels, isDefault = null, accountAlias, accountName }) {
     let channelNames = '';
     if (channels.length) {
       const channelsList = channels.map(({ channelId, channelName }) => `<#${channelId}|${channelName}>`).join(', ');
       channelName = ` This is the channel account for ${channelsList}`;
     }
     let isDefaultAccount = isDefault ? ' *This is the default workspace account.*' : '';
     return `• \`${accountAlias}\`: Slack alias for ${accountName}.${isDefaultAccount}${channelNames}\n`;
   }

   return allAccountsSafeView.map(mapFetchAccountsFromSafeView);
}